### ISSUE TITLE
Merged contraints lead to invalid type

### ISSUE THREAD

**Issue by frabbit** [HaxeFoundation/haxe#10446]
```haxe
class Test {
 static function main() {
   var x: { foo: Int, bar: Int, ohno: Int } = merge({ foo: 1}, { bar: 2});
 }
 
 static function merge<A:{}, B:{}, C:A & B>(a:A, b:B):C {
  return null;
 }
}
```

**Comment by frabbit** [comment]
see: https://github.com/HaxeFoundation/haxe/issues/9554#issuecomment-949964132

**Comment by frabbit** [comment]
this is not really a bug, but it shows that this technique is not a replacement for returning `A & B`.

**Comment by Simn** [comment]
I suppose we really do need `TIntersect` or something to that extent...

**Comment by frabbit** [comment]
Ah, i just realized that `&` is just `AExtend` for `AnonStatus`, right?

**Comment by Simn** [comment]
Yes. Originally I thought that there should be a way to directly handle `&` when translating from syntax to typed expressions, but this falls apart when type parameters get involved (and probably in other situations as well).
